_layouts/post.html

```diff
@@ -16,9 +16,21 @@
   {% include header.html %}
 
   <main>
-    <div class="post">
-      <h1>{{ page.title }}</h1>
+    <div class="post-header">
+      {% if page.header-image %}
+        <img class="post-header-image" src="{{ page.header-image | relative_url }}">
+      {% elsif page.thumbnail %}
+        <img class="post-header-image" src="{{ page.thumbnail | relative_url }}">
+      {% endif %}
+    </div>
 
+    {% if page.header-image or page.thumbnail %}
+      <div class="post post-body-header-border">
```

Comment:
I'd make the `post-body-header-border` slightly clearer, perhaps to `curved-header`. Note, we don't need to include the `post-` prefix as we're using SASS and can just use `.post.curved-header` in the SASS or whatever name you go with.